### PR TITLE
Draft create a pr

### LINKED ISSUE #17: As a reviewer, when I submit my transcript, a PR is opened on my behalf

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**
Closes #17

### CODE REVIEW

**Comment by adamjonas** [comment]
Getting this error:

Screen Shot 2023-04-15 at 3 03 58 PM

Full dump at https://gist.github.com/adamjonas/5166b98532b5eae42f0d9faec756b137#file-pr_error-json-L308

**Comment by adamjonas** [comment]
This is nice UI
Screen Shot 2023-04-15 at 8 57 22 PM

**Comment by adamjonas** [comment]
Still cannot open a PR on the newest transcript from the index page.

Also, left comments on the resulting [PR#104](https://github.com/adamjonas/bitcointranscripts/pull/104/files) and [#105](https://github.com/adamjonas/bitcointranscripts/pull/105/files#) on my repo. The metadata with the arrays is malformed.

File: src/pages/api/github/pr.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  const transcriptData = `${metadata}\n${transcribedText}\n`;
+  const fileContent = Buffer.from(transcriptData).toString("base64");
+
+  // Create new file on the branch
+  const path = directoryPath.split("/").slice(1).join("/");
+  const fileSlug = slugify(fileName, { remove: /[*+~.()'"!:@\\/]/g });
+  await octokit.request("PUT /repos/:owner/:repo/contents/:path", {
+    owner: forkOwner,
+    repo: forkRepo,
+    path: `${path}/${fileSlug}.md`,
```
Comment (adamjonas) [inline src/pages/api/github/pr.ts:64]: this does work for the How Bitcoin Works transcript.

The URL for the PUT is `https://api.github.com/repos/adamjonas/bitcointranscripts/contents/%2FHow-Bitcoin-Actually-Works-Episode-3-or-Hello-Bitcoin.md` and the error message is that ` message: 'path cannot start with a slash'`. There seems to be some UTF-8 encoding of `%2F`. I hard coded it for [#107](https://github.com/adamjonas/bitcointranscripts/pull/107/) to go through.

**Review by adamjonas — COMMENTED** [review]
Besides hardening up the path in the pr.ts. I left comments on the resulting [PR#104](https://github.com/adamjonas/bitcointranscripts/pull/104/files) and [#105](https://github.com/adamjonas/bitcointranscripts/pull/105/files#) and [#107](https://github.com/adamjonas/bitcointranscripts/pull/107/files#) on my repo. The metadata with the array values is malformed. When values of metadata fields are empty, they should not be added and things like speakers, tags and categories are arrays. See the bitcointranscripts [README](https://github.com/bitcointranscripts/bitcointranscripts#contribution) for reference.

File: src/pages/transcripts/[id].tsx [hunk 2]
```diff
@@ -94,12 +95,17 @@ const TranscriptPage = () => {
...
 
   const handleSubmit = async (editedContent: EditedContent) => {
+    const {editedDate, editedTags, editedSpeakers, editedCategories} = editedContent;
     setSubmitLoading(true);
     axios
       .post("/api/github/pr", {
         directoryPath: data?.content.loc ?? "bitcointranscripts/misc",
         fileName: "Sample Test Transcript",
         url: data?.content.media,
+        date: editedDate && dateFormat(editedDate),
```
Comment (adamjonas) [inline src/pages/transcripts/[id].tsx:129]: this isn't correct. This is the date of the recording.

File: src/utils.ts [hunk 3]
```diff
@@ -78,7 +79,7 @@ export class Metadata {
   public fileTitle: string;
   public source: string;
 
-  constructor(fileTitle: string, username: string, url: string, tags?: string[], speakers?: string[], categories?: string[]) {
+  constructor(fileTitle: string, username: string, url: string, date: string, tags?: string[], speakers?: string[], categories?: string[]) {
```
Comment (adamjonas) [inline src/utils.ts]: if `None` don't include them

**Review by nully0x — APPROVED** [review]

**Review by nully0x — COMMENTED** [review]
There is a conflict preventing rebasing. Is its ok to merge?

File: src/pages/api/github/pr.ts (hunk 1 above)
Comment (Emmanuel-Develops) [reply src/pages/api/github/pr.ts]: So I had the wrong format for the location path. I accounted for `"bitcointranscripts/"` prepended to the location whereas tstbtc doesn't prepend it hence `${path}` didn't exist and `"/"` was appended to the path which is why it kept failing. I updated the logic in 2c80bf8bba046582a50281589e908e4aac63526c

File: src/utils.ts (hunk 3 above)
Comment (Emmanuel-Develops) [reply src/utils.ts]: Done

**Comment by Emmanuel-Develops** [comment]
> There is a conflict preventing rebasing. Is its ok to merge?

yeah. It was the pr merged yesterday changing originalContent to content. I have the fix on this branch too do it's fine

File: src/utils.ts [hunk 4]
```diff
@@ -71,3 +73,118 @@ export const getCount = (item: number | string) => {
...
+    speakers,
+    categories,
+  }: MetadataProps) {
+    this.username = username;
+    this.fileTitle = fileTitle;
+    this.source = url;
+
+    // eslint-disable-next-line prettier/prettier
+    this.metaData =
+      `---\n` + `title: ${fileTitle}\n` + `transcript_by: ${username} \n`;
```
Comment (adamjonas) [inline src/utils.ts:98]: so this isn't quite right. We pass in who the transcript is by in the content payload. We should use that. We can create something new here like `reviewed_by`

**Comment by adamjonas** [comment]
One small thing on the transcript_by field. Otherwise, I'm good to merge it. Well done.

**Comment by adamjonas** [comment]
2 other things:

1. the file name still needs to be just alphanumeric:

https://github.com/adamjonas/bitcointranscripts/pull/109/files has a comma misc/James-Chiang,-Schnorr-and-Taproot-Chaincode-Residency-Demo-Day.md

2. Every newly created directory, needs a new _index.md file. It should look like this https://github.com/adamjonas/bitcointranscripts/pull/88/files. Is that built-in?

**Comment by Emmanuel-Develops** [comment]
closes #10

File: src/layout/index.tsx [hunk 5]
```diff
@@ -9,7 +9,7 @@ const Layout = ({ children }: { children: React.ReactNode }) => {
   return (
     <div className={styles.app_container}>
       <Head>
-        <title>Cryptoverse</title>
+        <title>Bitcoin Transcripts</title>
```
Comment (adamjonas) [inline src/layout/index.tsx:12]:
```suggestion
        <title>Bitcoin Transcription Reviews</title>
```

File: src/layout/index.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ const Layout = ({ children }: { children: React.ReactNode }) => {
...
         <meta name="description" content="Generated by create next app" />
```
Comment (adamjonas) [inline src/layout/index.tsx:13]:
```suggestion
        <meta name="description" content="Bitcoin Transcription Review" />
```

File: src/utils.ts [hunk 7]
```diff
@@ -71,3 +73,131 @@ export const getCount = (item: number | string) => {
...
+    speakers,
+    categories,
+  }: MetadataProps) {
+    this.fileTitle = fileTitle;
+    this.source = url;
+
+    // eslint-disable-next-line prettier/prettier
+    this.metaData = `---\n` +
+                    `title: ${fileTitle}\n` +
+                    `transcript_by: ${transcript_by} \n`;
```
Comment (adamjonas) [inline src/utils.ts:97]:
```suggestion
                    `transcript_by: ${transcript_by}\n`;
```
Is this space causing an extra line to be created? see https://github.com/adamjonas/bitcointranscripts/pull/114/files#r1170717938

File: src/utils.ts [hunk 8]
```diff
@@ -71,3 +73,131 @@ export const getCount = (item: number | string) => {
...
+    }
+
+    if (speakers) {
+      this.metaData += `speakers: ${speakers}\n`;
+    }
+
+    if (categories) {
+      this.metaData += `categories: ${categories}\n`;
+    }
+    this.metaData += `date: ${date}\n`;
```
Comment (adamjonas) [inline src/utils.ts:112]: this is not the correct date format. It should be `yyyy-md-dd` not `yyyy/mm/dd`

**Review by adamjonas — COMMENTED** [review]
getting close! let's get it in tomorrow.

File: src/utils.ts (hunk 8 above)
Comment (Emmanuel-Develops) [reply src/utils.ts:112]: alright

**Comment by BalogunofAfrica** [comment]
@Emmanuel-Develops  Since https://github.com/bitcointranscripts/transcription-review-front-end/pull/18/commits/2837297ccc8087b6e38a7ade3452e1a322862b94 closes https://github.com/bitcointranscripts/transcription-review-front-end/issues/19, I think you can also just change the favicon

**Comment by adamjonas** [comment]
For the record, this isn't working for me locally but the button currently does nothing on the main branch so I'm merging to get it in and deployed and if it needs some adjustments then so be it. Yay for no users!

**Pull request merged**